polygeist/lib/Conversion/PolygeistToLLVM/PolygeistToLLVM.cpp

```diff
@@ -485,7 +511,8 @@ struct TypeAlignOpLowering : public ConvertOpToLLVMPattern<TypeAlignOp> {
 void populatePolygeistToLLVMConversionPatterns(LLVMTypeConverter &converter,
                                                RewritePatternSet &patterns) {
   patterns.add<TypeSizeOpLowering, TypeAlignOpLowering, SubIndexOpLowering,
-               Memref2PointerOpLowering, Pointer2MemrefOpLowering>(converter);
+               Memref2PointerOpLowering, Pointer2MemrefOpLowering,
+               AddrSpaceCastOpLowering>(converter);
```

Comment:
I think this pattern won't work for non-bare pointers. You should only add it when using such option.